### DISCUSSION TITLE
[FR] Hotkey to switch between two tabs (MRU)

### DISCUSSION THREAD

**Discussion by megamorphg** [doublecmd/doublecmd#1167] [Suggestions]
Frequently I have over a dozen tabs open in one pane and I'd like to switch between the two tabs (without having to replace the opposite pane). Can we create a simple hotkey to switch between the two most recent tabs?

A more advanced version of this would be to show a temporary MRU pop-up (as in VS Code's or Firefox) where user can continue pressing tab to go further back in history and upon release it would switch to the selected tab.

**Comment by JonnyTech** [comment]
Does `ctrl+tab` not work for you?

**Comment by JonnyTech** [comment]
I just reread your report, you probably want `alt+0` instead. See https://doublecmd.github.io/doc/en/shortcuts.html for more keyboard shortcuts.

**Comment by megamorphg** [comment]
@JonnyTech `Alt+0` goes to the last-most tab open in the pane. I'd like something similar to MRU list in VS Code and other systems with a lot of tabs. Currently this hotkey doesn't exist nor does a feature request for it.

The closest to this is the "History" pane `Ctrl+H` but it lists all directories accessed (and will replace current directory with that directory)... not just currently open tabs in a recently used order.

**Comment by timur-g** [comment]
ctrl+tab goes forward thru tabs. That's like default Firefox behavior. But Firefox has _very useful_ option "**Ctrl+Tab cycles through tabs in recently used order**" which I would like to see in DC.  @alexx2000 can you please consider?

**Reply by vertigo220** [reply]
It's not just Firefox. Chrome/Brave does this by default (I think this is actually the _default_ behavior for Firefox, too). Notepad++ does this as well, and has, IMO, the best implementation (ctrl+tab actually opens a list of the tabs in the order of their most recent access, so quickly pressing it will switch to the most recent tab, but holding ctrl and pressing tab repeatedly will move down the list, then releasing ctrl will select the highlighted tab; up and down also move through the list).

IMO this behavior makes _much_ more sense and is far more useful than just cycling through the tabs in order. When they just cycle in order, I never even bother using this "feature" as it's completely useless. I guess with DC being dual-pane, one could move the two working tabs over to the other pane so they just switch back-and-forth, then move them back when done, but that's a very disjointed work flow.

**Comment by Skif-off** [comment]
[cm_ShowTabsList](https://doublecmd.github.io/doc/en/cmds.html#cm_ShowTabsList)? `Ctrl+Shift+A` by default.

Maybe [Lua](https://doublecmd.github.io/doc/en/lua.html) can help? An example of getting a list of tabs and switching between them can be found in [ChooseTab.lua](https://github.com/j2969719/doublecmd-plugins/blob/master/scripts/lua/ChooseTab.lua).

**Comment by timur-g** [comment]
If there is a way for Lua to make  `ctrl+tab` cycle between tabs, please explain. If not, I would really like to see it working like that. 
I use  ctrl+tab in other programs, including mentioned here Firefox and Notepad++ and I miss it a lot in DC, trying that unconciously.   
If that is an option, everyone will be happy. IDK why this is a discussion, I do not expect opposing voice.

**Reply by antodc** [reply]
You can set it as an hotkey for cm_NextTab

Configuration -> Options - Keys -> Hot keys
Then, write 'nexttab' in the filter and add/edit hotkey

**Comment by timur-g** [comment]
No. Ctrl-Tab is current for Next tab (as in following neighbour tab) and I do not want to set another hotkey for that I never use.
I want cm_CycleTab where I can put Ctrl+Tab.

**Reply by antodc** [reply]
Maybe I misunderstood something. What is the difference between going to the next tab and cycling through tabs? In my case, when you use ctrl+tab as a shortcut for cm_NextTab, it works in DC exactly the same as in Firefox.

**Reply by antodc** [reply]
Sorry, I had a problem with my browser and I did not see that there were already other messages above. Now I understand better. Apologize for the confusion.
